test/chi0.jl

```diff
@@ -0,0 +1,39 @@
+using DFTK
+using Test
+using LinearAlgebra: norm
+
+include("testcases.jl")
+
+@testset "Computing χ0" begin
+    Ecut=3
+    fft_size = [10, 10, 1]
+    tol=1e-10
+    ε = 1e-8
+    kgrid = [3, 1, 1]
+    testcase = silicon
+    n_bands = 10
+
+    spec = ElementPsp(testcase.atnum, psp=load_psp(testcase.psp))
+    for temperature in (0, 0.03)
+        model = model_LDA(testcase.lattice, [spec => testcase.positions], temperature=temperature)
+
+        basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid, fft_size=fft_size, enable_bzmesh_symmetry=false)
+
+        ρ0 = guess_density(basis)
+        energies, ham = energy_hamiltonian(basis, nothing, nothing;
+                                           ρ=ρ0)
+        V = DFTK.total_local_potential(ham)
+        ρV = DFTK.next_density(ham, tol=tol, eigensolver=diag_full, n_bands=n_bands).ρ
+        χ0 = compute_χ0(ham)
+
+        # now we go change the local potential of the hamiltonian
+        # TODO this is a bit of a hack...
```

Comment:
yeah I thought about that, but then I thought we'd want the capability to change the local potential anyway. Maybe that was wrong?

Reply:
For testing it might not be a bad idea, but otherwise ... would you not just add an extra term? This `set_total_local_potential!` really bothers me and the fact that we need to play tricks over the `k`-Points might point into the direction that maybe we should not support it. Can you imagine a scenario where you really want to modify the total local potential, where it is not justified to construct a new "model", like a computational routine or algorithm?

Reply:
Hey that actually works. I thought about doing that but decided not to because I was thinking it'd be tricky to give a discretized local potential, but actually it's trivial, so perfect! This is way better, no bad mutation anymore.